src/main/java/net/sf/jabref/sql/database/MySQL.java

```diff
@@ -0,0 +1,135 @@
+                    + "PRIMARY KEY (strings_id) \n" + ");";
+        case GROUP_TYPES:
+            return "CREATE TABLE IF NOT EXISTS group_types ( \n"
+                    + "group_types_id  INTEGER     NOT NULL AUTO_INCREMENT, \n"
+                    + "label   VARCHAR(100)    DEFAULT NULL, \n"
+                    + "PRIMARY KEY (group_types_id) \n" + ");";
+        case GROUPS:
+            return
+                    "CREATE TABLE IF NOT EXISTS groups ( \n"
+                            + "groups_id       INTEGER         NOT NULL AUTO_INCREMENT, \n"
+                            + "group_types_id  INTEGER         DEFAULT NULL, \n"
+                            + "label           VARCHAR(100)    DEFAULT NULL, \n"
+                            + "database_id INT UNSIGNED, \n"
+                            + "parent_id       INTEGER         DEFAULT NULL, \n"
+                            + "search_field       VARCHAR(100)          DEFAULT NULL, \n"
+                            + "search_expression  VARCHAR(200)          DEFAULT NULL, \n"
+                            + "case_sensitive  BOOL          DEFAULT NULL, \n"
+                            + "reg_exp BOOL DEFAULT NULL, \n"
+                            + "hierarchical_context INTEGER DEFAULT NULL, \n"
+                            + "FOREIGN KEY (database_id) REFERENCES jabref_database(database_id), \n"
+                            + "PRIMARY KEY (groups_id) \n" + ");";
+        case ENTRY_GROUP:
+            return
+                    "CREATE TABLE IF NOT EXISTS entry_group ( \n"
+                            + "entries_id       INTEGER        NOT NULL AUTO_INCREMENT, \n"
+                            + "groups_id        INTEGER        DEFAULT NULL, \n"
+                            + "INDEX(entries_id), \n"
+                            + "INDEX(groups_id), \n"
+                            + "FOREIGN KEY (entries_id) REFERENCES entries(entries_id) ON DELETE CASCADE, \n"
+                            + "FOREIGN KEY (groups_id)  REFERENCES groups(groups_id), \n"
+                            + "PRIMARY KEY (groups_id, entries_id) \n" + ");";
+        default:
+            return "";
+        }
+
+    }
+
+    private static final String OPT_ALLOW_MULTI_QUERIES = "?allowMultiQueries=true";
+
+    @Override
+    public Connection connectAndEnsureDatabaseExists(DBStrings dbStrings)
+            throws SQLException, IllegalAccessException, ClassNotFoundException, InstantiationException {
+
+        dbStrings.setDbParameters(OPT_ALLOW_MULTI_QUERIES);
+        String url = SQLUtil.createJDBCurl(dbStrings, false);
+
+        Connection conn = connect(url, dbStrings.getDbPreferences().getUsername(), dbStrings.getPassword());
+
+        SQLUtil.processQuery(conn, "CREATE DATABASE IF NOT EXISTS `" + dbStrings.getDbPreferences().getDatabase() + '`');
+        conn.setCatalog(dbStrings.getDbPreferences().getDatabase());
```

Comment:
Just one thing I got in mind while looking through an old PR I created a while ago:
There is a potential resource leak in here:
If the Create Database query fails and throws an SQLException, the connection does not get closed and remains open.
That was  one thing I fixed in my old PR #910 because it was discovered from coverity amalysis.
My solution was to catch the exception and close the conn. Then re throw the ex.
https://github.com/Siedlerchr/jabref/blob/reworkedCoverityFixes/src/main/java/net/sf/jabref/sql/exporter/MySQLExporter.java